b2sdk/account_info/abstract.py

```diff
@@ -218,23 +230,21 @@ def set_auth_data(
         :param str realm: a realm to authorize account in
         :param dict allowed: the structure to use for old account info that was saved without 'allowed'
         :param str application_key_id: application key ID
+        :param str s3_api_url: S3-compatible API URL
 
         .. versionchanged:: 0.1.5
            `account_id_or_app_key_id` renamed to `application_key_id`
         """
         if allowed is None:
             allowed = self.DEFAULT_ALLOWED
         assert self.allowed_is_valid(allowed)
+
+        if s3_api_url is None:
```

Comment:
I don't understand why the whole `v2` api change thing - we can make `s3_api_url` optional param (here we are checking for `None`) - `_set_auth_data` is private interface so don't have to support it...